crates/evm/traces/src/decoder/mod.rs

```diff
@@ -241,28 +206,37 @@ impl CallTraceDecoder {
         }
     }
 
+    /// Decodes all nodes in the specified call trace.
     pub async fn decode(&self, traces: &mut CallTraceArena) {
         for node in &mut traces.arena {
             // Set contract name
-            if let Some(contract) = self.contracts.get(&node.trace.address).cloned() {
-                node.trace.contract = Some(contract);
+            if let Some(contract) = self.contracts.get(&node.trace.address) {
+                node.trace.contract = Some(contract.clone());
             }
 
             // Set label
-            if let Some(label) = self.labels.get(&node.trace.address).cloned() {
-                node.trace.label = Some(label);
+            if let Some(label) = self.labels.get(&node.trace.address) {
+                node.trace.label = Some(label.clone());
             }
 
+            // Decode events
+            self.decode_events(node).await;
+
             // Decode call
-            if let Some(precompile_fn) = self.precompiles.get(&node.trace.address) {
-                node.decode_precompile(precompile_fn, &self.labels);
-            } else if let RawOrDecodedCall::Raw(ref bytes) = node.trace.data {
-                if bytes.len() >= 4 {
+            // TODO: chain ID argument
```

Comment:
This is to decode different precompiles per chain id?

Reply:
Yes that's what I was thinking, but maybe I'm thinking ahead too much. Maybe we should just try to decode whatever revm has set as precompiles

Reply:
Yeah I think might be best to stick to what `revm` has—I think soft consensus is to not get into maintaining precompiles for other chains—they can become a pain. IIRC arbitrum, which is the biggest offender, has a repo that simulates their custom precompile that ppl can use

Reply:
Ref https://github.com/foundry-rs/foundry/issues/748